PWGJE/Core/JetTaggingUtilities.h

```diff
@@ -99,94 +99,109 @@ int getOriginalMotherIndex(const typename T::iterator& particle)
 
 /**
  * returns the globalIndex of the earliest HF mother of a particle in the shower. returns -1 if a suitable mother is not found. Should be used only on already identified HF particles
+ * add priority to select HF shower based on status
  *
  * @param hfparticle MCParticle whose mother is to be found
+ * @param woGluonShower
  */
 
 template <typename T>
-int getOriginalHFMotherIndex(const typename T::iterator& hfparticle)
+int getOriginalHFMotherIndex(const typename T::iterator& hfparticle, bool woGluonShower)
 {
+  auto statusPriority = [](int status) -> int {
+    switch (status) {
+      case 33:
+        return 1; // High priority
+      case 43:
+        return 2;
+      case 23:
+        return 3;
+      case 63:
+        return 4;
+      case 51:
+        return 5;
+      default:
+        return 6;
```

Comment:
Don't use magic numbers. See the [coding guidelines](https://rawgit.com/AliceO2Group/CodingGuidelines/master/coding_guidelines.html?showone=Magic_numbers#Magic_numbers).
Define an enumerator with names for these values.